feature/experimental/qos/qos_ingress_traffic_classification/ate_tests/ingress_traffic_classification_test/ingress_traffic_classification_test.go

```diff
@@ -0,0 +1,1089 @@
+
+	"github.com/openconfig/featureprofiles/internal/deviations"
+	"github.com/openconfig/featureprofiles/internal/fptest"
+	gpb "github.com/openconfig/gnmi/proto/gnmi"
+	"github.com/openconfig/ondatra"
+	"github.com/openconfig/ondatra/gnmi"
+	"github.com/openconfig/ondatra/gnmi/oc"
+	"github.com/openconfig/ygot/ygot"
+)
+
+type trafficData struct {
+	trafficRate float64
+	frameSize   uint32
+	codePoint   uint8
+	queue       string
+}
+
+func TestMain(m *testing.M) {
+	fptest.RunTests(m)
+}
+
+// Test cases:
+//
+// Verify the presence of the telemetry paths of the following features:
+//  -  Configure DUT with ingress and egress routed interfaces.
+//  -  Configure QOS classifiers to match packets arriving on DUT ingress interface to corresponding forwarding class according to
+//     classification table.
+//  -  Configure packet re-marking for configured classes according to the marking table.
+//  -  One-by-one send flows containing every TOS/TC/EXP value in the classification table.
+//  -  For every flow sent, verify match-packets counters on the DUT ingress interface
+//  -  Verify packet markings on ATE ingress interface.
+//  -  Verify that no traffic drops in all flows
+//  - Check the QoS queue counters exist and are updated correctly
+//    - /qos/interfaces/interface/output/queues/queue/state/transmit-pkts
+//    - /qos/interfaces/interface/output/queues/queue/state/transmit-octets
+//    - /qos/interfaces/interface/output/queues/queue/state/dropped-pkts
+//    - /qos/interfaces/interface/output/queues/queue/state/dropped-octets
+//    - /qos/interfaces/interface/input/classifiers/classifier/terms/term/state/matched-packets
+//    - /qos/interfaces/interface/input/classifiers/classifier/terms/term/state/matched-octets
+//
+// Topology:
+//   ate:port1 <--> port1:dut:port2 <--> ate:port2
+//
+//  Sample CLI command to get telemetry using gnmic:
+//   - gnmic -a ipaddr:50051 -u username -p password --skip-verify get \
+//      --path /components/component --format flat
+//   - gnmic tool info:
+//     - https://github.com/karimra/gnmic/blob/main/README.md
+//
+
+func TestQoSClassifier(t *testing.T) {
+	dut := ondatra.DUT(t, "dut")
+	dp1 := dut.Port(t, "port1")
+	dp2 := dut.Port(t, "port2")
+
+	// Configure DUT interfaces and QoS.
+	configureDUTIntf(t, dut)
+	configureQoS(t, dut)
+
+	// Configure ATE interfaces.
+	ate := ondatra.ATE(t, "ate")
+	ap1 := ate.Port(t, "port1")
+	ap2 := ate.Port(t, "port2")
+	top := ate.Topology().New()
+	intf1 := top.AddInterface("intf1").WithPort(ap1)
+	intf1.IPv4().
+		WithAddress("198.51.100.1/31").
+		WithDefaultGateway("198.51.100.0")
+	intf1.IPv6().
+		WithAddress("2001:db8::2/126").
+		WithDefaultGateway("2001:db8::1")
+	intf2 := top.AddInterface("intf2").WithPort(ap2)
+	intf2.IPv4().
+		WithAddress("198.51.100.3/31").
+		WithDefaultGateway("198.51.100.2")
+	intf2.IPv6().
+		WithAddress("2001:db8::6/126").
+		WithDefaultGateway("2001:db8::5")
+	top.Push(t).StartProtocols(t)
+
+	type qosVals struct {
```

Comment:
Should be able to use the entity naming libraries `CommonTrafficQueues()` function.

Reply:
ANCX queues are different to CommonTrafficQueues(), can we update the CommonTrafficQueues() to include these queue mappings as well ?

Reply:
Aha, thanks. I will work to make this function (or a different one) work for the ancx use case, too. You can leave as-is for now.